### DISCUSSION TITLE
Subsetting peak matrix by bed file

### DISCUSSION THREAD

**Discussion by LacquerHed** [GreenleafLab/ArchR#1828] [Questions / Documentation]
I am interested in subsetting the peakmatrix by some peaks that I validated in tools outside R, wondering how I can do that? I essentially just have an external bed file of these peak locations, would need to match them to their complete information in the total ArchR peak matrix and cut out everything else. Thanks.

**Comment by rcorces** [comment]
`addPeakSet()` and then re-run `addPeakMatrix()`

**Reply by LacquerHed** [reply]
Thanks so much this worked well, the part where Im running into problems is foot-printing some TFs using only this specific peakset? I tried the following:

```
motifPositions <- getPositions(peak_subset)

motifPositions

motifs <- c("Klf6")
markerMotifs <- unlist(lapply(motifs, function(x) grep(x, names(motifPositions), value = TRUE)))
markerMotifs

#get footprints
seFoot <- getFootprints(
  ArchRProj = peak_subset, 
  useGroups = c("5","0"),
  positions = motifPositions[markerMotifs], 
  groupBy = "Clusters2"
)
```

However it does not limit the foot-printing to only that peak-set but seems to do the whole peak matrix even though I only have one peak-set in this object as seen below, not sure where I am going wrong:

```
peak_subet@peakSet

GRanges object with 171 ranges and 5 metadata columns:
        seqnames            ranges strand |        name     score       idx        GC         N
           <Rle>         <IRanges>  <Rle> | <character> <numeric> <integer> <numeric> <numeric>
    [1]     chr1 36255029-36255082      * |  Peak_22429        62         4    0.6296         0
    [2]     chr1 43474470-43474576      * |  Peak_80160        18         8    0.4953         0
    [3]     chr1 74295441-74295933      * |   Peak_4031       194         1    0.5639         0
    [4]     chr1 88474822-88474954      * |   Peak_8888       123         2    0.4737         0
    [5]     chr1 88639805-88639807      * |  Peak_71736        20         7    0.6667         0
    ...      ...               ...    ... .         ...       ...       ...       ...       ...
  [167]    chr19 28624754-28624976      * |  Peak_58558        24         9    0.4529         0
  [168]    chr19 38038164-38038406      * |  Peak_32271        44         5    0.4321         0
  [169]    chr19 57079308-57079576      * |  Peak_55793        26         7    0.4349         0
  [170]    chr19 57079777-57079808      * | Peak_100159        13        11    0.5938         0
  [171]    chr19 60939907-60940407      * |  Peak_12905        96         4    0.4631         0
  -------
  seqinfo: 19 sequences from an unspecified genome; no seqlengths
```

Thanks again.

**Reply by rcorces** [reply]
I dont follow your post. Footprinting is performed using group coverage files, not the peak matrix?
What defines the locations where footprinting will be performed is `motifPositions[markerMotifs]`. If that looks correct, then it should only be footprinting using those locations. If you feel that isnt the case, please create a reproducible example using the tutorial data.

**Reply by LacquerHed** [reply]
Thats where I am a bit confused, I used your previous advice to add an external peakSet (via a GRanges object from an imported bed file) followed by re-adding peakMatrix, however when I scan motif positions in this new ArchRproject it still gives me the same number as that of the original ArchRproject which the full initial peak-set and initial peak-matrix. I am trying to find motif positions using this new limited ArchRproject with novel peak-set and re-added peak-matrix.
Thanks.

**Reply by LacquerHed** [reply]
Actually I have figured out my issue, the peakset formatting of my imported BED doesnt match the ArchR GRanges peakset format. You can close this issue, thank you for your help again.

**Reply by rcorces** [reply]
did you re-add the motif annotations after updating your peak set using `force = TRUE`?

**Reply by LacquerHed** [reply]
Yes I figured out how to get to the point of calling up/down motifs on new peak-set after all of these steps however got the following error:

```
Error in peakAnnoEnrichment(seMarker = markerTest, ArchRProj = PEC_0_enahncer_peaks,  : 
  Peaks from matches do not match peakSet in ArchRProj!
```

I know it stems from my imported peak-set being in the wrong format, because externally when I mess with it using bedtools/bedops etc I use a scaled down version and when re-imported that does not work. So I think this should solve the problem, thanks again.

**Comment by LacquerHed** [comment]
When adding a peak-set to an ArchRproject, does it have to be in identical format to peak-sets called within ArchR? The issue Im running into is that I use getMarkerFeatures to find some differential peaks and then export this result for external manipulations outside of RStudio. Upon reimporting it, wondering how I can add this as a peak-set that will work for all downstream ArchR functions? Such as peakAnnoEnrichment etc. Thanks.

**Reply by rcorces** [reply]
Please create a reproducible example using the tutorial dataset if you want help. It is extremely difficult to help without this.

**Reply by LacquerHed** [reply]
I will def do this, but very simplified, what format does the imported and added peak-set have to be in to work for all downstream ArchR functions? Thanks.

**Reply by rcorces** [reply]
It shouldnt really matter as long as it is a GRanges object. It is possible that there is a bug in the code where a user-supplied peak set isnt properly annotated for footprinting. So I would just simulate your current workflow on the tutorial data and see if it results in the same problem in a way that I would be able to reproduce to troubleshoot
